bin/propolis-server/src/lib/spec.rs

```diff
@@ -14,22 +13,13 @@ use crate::config;
 
 /// Errors that can occur while building an instance spec from component parts.
 #[derive(Debug, Error)]
-pub enum SpecBuilderError {
-    #[error("A device with name {0} already exists")]
-    DeviceNameInUse(String),
-
-    #[error("A backend with name {0} already exists")]
-    BackendNameInUse(String),
-
-    #[error("A PCI device is already attached at {0:?}")]
-    PciPathInUse(PciPath),
+pub enum ServerSpecBuilderError {
+    #[error("Interior spec builder returned an error: {0}")]
+    InnerBuilderError(#[from] SpecBuilderError),
```

Comment:
`#[error(transparent)]` might be suitable here if we just want to forward the `SpecBuilderError`s

Reply:
Good call. Fixed in b243962.